### ISSUE TITLE
Error populating saved model

### ISSUE THREAD

**Issue by sr33kant** [clab/dynet#1587]
It could be me as im still getting familiar with the framework. My class init has all the parameters initialized and I have a save method to save the model to a file. I'm trying to populate the file created earlier 
parser.Save("C:\\saved_models\\relExtract.model")
infer_model=dynet.ParameterCollection()
p1=parser.load("C:\\saved_models\\relExtract.model",infer_model)

My learner class is structured like this with trainer , predict , save and load methods

class relExtract:
          def __init__(self, vocab, ner, some_arg, some2_arg, w2i, c2i, options):
                 self.model = ParameterCollection()
                 random.seed(1)
       
                 self.trainer=AdagradTrainer(self.model)
I have vanilla LSTM layers , would that cause issues with retrieving/saving model? Currently Im unable to load the model that is being saved

FIrst couple of lines of the model file that is getting saved

#Parameter# /vanilla-lstm-builder/_0 {400,150} 960001 ZERO_GRAD
#Parameter# /vanilla-lstm-builder/_2 {400} 6401 ZERO_GRAD
